docs/source/developer_guide/guides/3_simple_cpp_stage.md

```diff
@@ -53,34 +53,36 @@ template <typename SourceT>
 class PythonSource : ...
 ```
 
-The `SourceT` type will be the datatype emitted by this stage, which is not required to be a `shared_ptr` to a Morpheus message. In contrast, general stages and sinks must inherit from the `PythonNode` class:
+The `SourceT` type will be the datatype emitted by this stage. In contrast, general stages and sinks must inherit from the `PythonNode` class, which specifies both receive and emit types:
 
 ```cpp
-template <typename SinkT, typename SourceT = SinkT>
+template <typename SinkT, typename SourceT = SinkT>  // by default, emit type == receive type
```

Comment:
I thought the `SourceT = SinkT` part was a default value for SourceT such that if it (a type for SourceT) was not provided, it would default to the same type as SinkT. [reference](https://docs.microsoft.com/en-us/cpp/cpp/templates-cpp?view=msvc-170#default-template-arguments) -- but I'm new to templating in C++ so I could 100% be wrong.

Reply:
oh! You are right, good catch.